prototype_source/maskedtensor_adagrad.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+
+"""
+(Prototype) Efficiently writing "sparse" semantics for Adagrad with MaskedTensor
+================================================================================
+"""
+
+######################################################################
+# Before working through this tutorial, please review the MaskedTensor
```

Comment:
nice

Reply:
tho the links don't currently work --i'm assuming they're referring to the pages that haven't yet landed

Reply:
yep - the output won't render anyway since the link only renders html, so I can guarantee the formatting, just need a check on the wording